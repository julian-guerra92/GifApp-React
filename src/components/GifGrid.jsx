import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

   const { images, isLoading } = useFetchGifs(category);

   return (
      <>
         <h3>{category}</h3>
         {
            isLoading && (<h2>Cargando...</h2>)
         }
         <div className='card-grid'>
            {
               images.map((image) => (
                  <GifItem
                     key={image.id}
                     {...image}
                  />
               ))
            }
         </div>
      </>
   )
}

/*
 - El useEffect hace que los elementos especificados se rendericen una única vez cuando el componente es creado. Así se evita renderizar o hacer peticiones nuevamente de funcionalidades de nuestra aplicación.
*/