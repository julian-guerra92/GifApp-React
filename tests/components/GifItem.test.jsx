import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Prueba sobre el GifItem Component', () => {

   const title = 'one punch GIF by Adult Swim';
   const url = 'https://giphy.com/gifs/adultswim-adult-swim-toonami-l3vRlt8kty8KKeHni'

   test('Debe de hacer math con el snapshot', () => {
      const { container } = render(<GifItem title={title} url={url} />);
      expect(container).toMatchSnapshot();
   })

   test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
      render(<GifItem title={title} url={url} />);
      const {src, alt} = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
   })

   test('Debe de mostrar el tÃ­tulo en el componente', () => {
      render(<GifItem title={title} url={url} />);
      expect(screen.getByText(title)).toBeTruthy();
   })

})