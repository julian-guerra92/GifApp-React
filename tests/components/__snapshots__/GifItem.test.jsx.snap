// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el GifItem Component Debe de hacer math con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch GIF by Adult Swim"
      src="https://giphy.com/gifs/adultswim-adult-swim-toonami-l3vRlt8kty8KKeHni"
    />
    <p>
      one punch GIF by Adult Swim
    </p>
  </div>
</div>
`;
